hack/ginkgo-e2e.sh

```diff
@@ -34,9 +34,12 @@ e2e_test=$(kube::util::find-binary "e2e.test")
 GINKGO_PARALLEL=${GINKGO_PARALLEL:-n} # set to 'y' to run tests in parallel
 CLOUD_CONFIG=${CLOUD_CONFIG:-""}
 
-# If 'y', Ginkgo's reporter will not print out in color when tests are run
-# in parallel
-GINKGO_NO_COLOR=${GINKGO_NO_COLOR:-n}
+# If 'y', Ginkgo's reporter will not use escape sequence to color output.
+#
+# Since Kubernetes 1.25, the default is to use colors only when connected to
+# a terminal. That is the right choice for all Prow jobs (Spyglass doesn't
+# render them properly).
+GINKGO_NO_COLOR=${GINKGO_NO_COLOR:-$(if [ -t 2 ]; then echo n; else echo y; fi)}
```

Comment:
TIL https://unix.stackexchange.com/a/401938

why only on 2? and not 0? any special reason

Reply:
Because we care about the output, not the input.

Reply:
(And because stderr is where fancy diagnostic output _should_ be going as opposed to structured program output to stdout, so stderr not stdout)